### ISSUE TITLE
Investigate for NUnit Version 4 support

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#596]
We should investigate whether our current analyzers and codefixes for NUnit 3 also work with NUnit Version 4.

- https://docs.nunit.org/articles/nunit/Towards-NUnit4.html
- Release notes as PR: https://github.com/nunit/nunit/issues/3325

**Comment by RalfKoban** [comment]
NUnit 4 was released yesterday.

It contains some breaking changes:
https://docs.nunit.org/articles/nunit/release-notes/breaking-changes.html#nunit-40

As well as an migration guide:
https://docs.nunit.org/articles/nunit/release-notes/Nunit4.0-MigrationGuide.html


When trying to use that, some of our tests for following rules fail:
- MiKo_3105
- MiKo_3106
- MiKo_3110
- MiKo_3113
- MiKo_6002

This is because the test data contains now code that is no longer available (e.g. the classic assertions have been moved into a different project and namespace `NUnit.Framework.Legacy`.

We should detect those as well and especially our codefix providers should provide correct fixes.

For example, the format as message incl. its arguments was removed, such as:
```c#
Assert.IsTrue(flag, "some message {0}", someArgument);
```

It got replaced by formattable strings. So our codefix providers should fix them into something like:
```c#
Assert.That(flag, Is.True, $"some message {someArgument}");
```
